x/concentrated-liquidity/lp.go

```diff
@@ -47,19 +47,24 @@ func (k Keeper) createPosition(ctx sdk.Context, poolId uint64, owner sdk.AccAddr
 	initialSqrtPrice := pool.GetCurrentSqrtPrice()
 	initialTick := pool.GetCurrentTick()
 
+	// Calculate the amount of liquidity that will be added to the pool by creating this position.
+	liquidityDelta := math.GetLiquidityFromAmounts(pool.GetCurrentSqrtPrice(), sqrtPriceLowerTick, sqrtPriceUpperTick, amount0Desired, amount1Desired)
+	if liquidityDelta.IsZero() {
+		return sdk.Int{}, sdk.Int{}, sdk.Dec{}, errors.New("liquidityDelta calculated equals zero")
+	}
+
 	// If the current square root price and current tick are zero, then this is the first position to be created for this pool.
 	// In this case, we calculate the square root price and current tick based on the inputs of this position.
-	if k.isInitialPosition(initialSqrtPrice, initialTick) {
-		err := k.initializeInitialPosition(cacheCtx, pool, amount0Desired, amount1Desired)
-		if err != nil {
+	if k.isInitialPositionForPool(initialSqrtPrice, initialTick) {
+		// initialize fee accumulator position
+		if err := k.initializeFeeAccumulatorPosition(cacheCtx, poolId, owner, liquidityDelta); err != nil {
 			return sdk.Int{}, sdk.Int{}, sdk.Dec{}, err
 		}
-	}
 
-	// Calculate the amount of liquidity that will be added to the pool by creating this position.
-	liquidityDelta := math.GetLiquidityFromAmounts(pool.GetCurrentSqrtPrice(), sqrtPriceLowerTick, sqrtPriceUpperTick, amount0Desired, amount1Desired)
-	if liquidityDelta.IsZero() {
-		return sdk.Int{}, sdk.Int{}, sdk.Dec{}, errors.New("liquidityDelta calculated equals zero")
+		err := k.initializeInitialPositionForPool(cacheCtx, pool, amount0Desired, amount1Desired)
+		if err != nil {
+			return sdk.Int{}, sdk.Int{}, sdk.Dec{}, err
+		}
```

Comment:
Upon more testing, I found that to be correct, the call to `initializeFeeAccumulatorPosition` must be surrounded by the call to `hasPosition` instead of `isInitialPositionForPool`.

I updated this branch to reflect this: https://github.com/osmosis-labs/osmosis/pull/3893/files#diff-172bed5666a1e4a35522b031d46c6c7af3380aa208b753f336635ac046f1af1f

The reason is that `isInitialPosiitonForPool` returns true if this is the first position for pool in general. However, we want to call `initializeFeeAccumulatorPosition` only if this is the first time a given owner is adding liquidity for certain ticks. `hasPosition` does exactly that.

In other words, this is what I'm suggesting: https://github.com/osmosis-labs/osmosis/blob/5531b3f9e9a937eb1705d239c8ec18516ee6b4c4/x/concentrated-liquidity/lp.go#L65-L69

Please let me know if the suggestion makes sense

Reply:
passing this for now with agreement that we will becoming back to lp.go changes in the future